packages/grafana-prometheus/src/querybuilder/components/MetricCombobox.tsx

```diff
@@ -0,0 +1,125 @@
+import { useCallback } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { EditorField, EditorFieldGroup } from '@grafana/experimental';
+import { InlineField, InlineFieldRow } from '@grafana/ui';
+import { Combobox, ComboboxOption } from '@grafana/ui/src/components/Combobox/Combobox';
+
+import { PrometheusDatasource } from '../../datasource';
+import { regexifyLabelValuesQueryString } from '../parsingUtils';
+import { QueryBuilderLabelFilter } from '../shared/types';
+import { PromVisualQuery } from '../types';
+
+export interface MetricComboboxProps {
+  metricLookupDisabled: boolean;
+  query: PromVisualQuery;
+  onChange: (query: PromVisualQuery) => void;
+  onGetMetrics: () => Promise<SelectableValue[]>;
+  datasource: PrometheusDatasource;
+  labelsFilters: QueryBuilderLabelFilter[];
+  onBlur?: () => void;
+  variableEditor?: boolean;
+}
+
+export function MetricCombobox({
+  datasource,
+  query,
+  onChange,
+  onGetMetrics,
+  labelsFilters,
+  variableEditor,
+}: Readonly<MetricComboboxProps>) {
+  /**
+   * Gets label_values response from prometheus API for current autocomplete query string and any existing labels filters
+   */
+  const getMetricLabels = useCallback(
```

Comment:
Do we have some debounce on this inside the combobox or no? If I remember having a gazillion metrics was a real usecase and even though we may display it fine in the combobox now, doing such request on every key stroke probably isn't a good idea.

Reply:
yes (soon) https://github.com/grafana/grafana/pull/95485